schemas/api/WCP6Goodbye.schema.json

```diff
@@ -0,0 +1,35 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://fdc3.finos.org/schemas/next/api/WCP6Goodbye.schema.json",
+  "title": "Web Connection Protocol 6 Goodbye",
+  "description": "Goodbye message to be sent to the Desktop Agent when disconnecting (e.g. when closing the window or navigating). Desktop Agents should close the MessagePort after receiving this message, but retain instance details in case the application reconnects (e.g. after a navigation event).",
+  "type": "object",
+  "allOf": [
+    {
+      "$ref": "#/$defs/WCP6GoodbyeBase"
+    },
+    {
+      "$ref": "WCPConnectionStep.schema.json"
```

Comment:
It doesn't really make sense to extend the connection base, as this means the shape of the object is this:

```
export interface WebConnectionProtocol6Goodbye {
    meta: ConnectionStepMetadata;
    /**
     * The message payload, containing data pertaining to this connection step.
     */
    payload: { [key: string]: any };
    /**
     * Identifies the type of the connection step message.
     */
    type: "WCP6Goodbye";
}
```

And `ConnectionStepMeta` has details of the connection being made.  But, all this is being passed down the message port, so the DA and Proxy both know who they're talking to.

Reply:
```
Comparing Validation failed: {
  "type": "WCP6Goodbye",
  "meta": {
    "requestUuid": "22c48242-8f16-4c4e-8159-ea0e1af6156e",
    "timestamp": "2024-09-17T08:15:06.159Z",
    "source": {
      "appId": "cucumber-app",
      "instanceId": "cucumber-instance"
    }
  },
  "payload": {}
} 
 [{"instancePath":"/meta","schemaPath":"#/required","keyword":"required","params":{"missingProperty":"connectionAttemptUuid"},"message":"must have required property 'connectionAttemptUuid'"}]
```

Reply:
I think it _does_ make sense to derive all connection messages from the same type as this enables TypeScript types that cover all the connection messages. Yes it leads to an unused payload property - but sending one (and only one) message without that property will no doubt trip someone else up in future.

If we moved it into the DACP (where I don't think it fits), it would still derive from a type with `payload` (agentRequest) and again it wouldn't make sense to have it be the only message without.

I take the point about not needing to quote the `connectionAttemptUuid`, which for some reason your example has down as `requestUuid`. Also, you have a `source` field that is not allowed by the schema (and wouldn't be needed for the same reason that the `connectionAttemptUuid` isn't). I can deal with the connectionAttemptUuid by moving the requirement into each type that really needs it - but does add some complexity to schema and probably generated types.

I've tried to give you what you want here, I think its added complexity to little benefit personally. See:  743ede1fdbe660dfec01df5c528ffc272cc434cf

Because of the additional complexity in the schema composition generated code has weirder naming - but then you don't have the payload property or connectionAttemptUUID on one message.

Reply:
Have a look - easy enough to revert.